plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/TestJdbcConnectionCreation.java

```diff
@@ -0,0 +1,116 @@
+import static io.trino.tpch.TpchTable.REGION;
+import static java.util.Objects.requireNonNull;
+
+@Test(singleThreaded = true) // inherited from BaseJdbcConnectionCreationTest
+public class TestJdbcConnectionCreation
+        extends BaseJdbcConnectionCreationTest
+{
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        String connectionUrl = createH2ConnectionUrl();
+        DriverConnectionFactory delegate = new DriverConnectionFactory(new Driver(), connectionUrl, new Properties(), new EmptyCredentialProvider());
+        this.connectionFactory = new ConnectionCountingConnectionFactory(delegate);
+        return createH2QueryRunner(ImmutableList.of(NATION, REGION), ImmutableMap.of("connection-url", connectionUrl), new TestingConnectionH2Module(connectionFactory));
+    }
+
+    @Test(dataProvider = "testCases")
+    public void testJdbcConnectionCreations(@Language("SQL") String query, int expectedJdbcConnectionsCount, Optional<String> errorMessage)
+    {
+        assertJdbcConnections(query, expectedJdbcConnectionsCount, errorMessage);
+    }
+
+    @DataProvider
+    public Object[][] testCases()
+    {
+        return new Object[][] {
+                {"SELECT * FROM nation LIMIT 1", 3, Optional.empty()},
+                {"SELECT * FROM nation ORDER BY nationkey LIMIT 1", 3, Optional.empty()},
+                {"SELECT * FROM nation WHERE nationkey = 1", 3, Optional.empty()},
+                {"SELECT avg(nationkey) FROM nation", 3, Optional.empty()},
+                {"SELECT * FROM nation, region", 6, Optional.empty()},
+                {"SELECT * FROM nation n, region r WHERE n.regionkey = r.regionkey", 6, Optional.empty()},
+                {"SELECT * FROM nation JOIN region USING(regionkey)", 6, Optional.empty()},
+                {"SELECT * FROM information_schema.schemata", 1, Optional.empty()},
+                {"SELECT * FROM information_schema.tables", 1, Optional.empty()},
+                {"SELECT * FROM information_schema.columns", 5, Optional.empty()},
+                {"SELECT * FROM nation", 3, Optional.empty()},
+                {"CREATE TABLE copy_of_nation AS SELECT * FROM nation", 11, Optional.empty()},
+                {"INSERT INTO copy_of_nation SELECT * FROM nation", 12, Optional.empty()},
+                {"DELETE FROM copy_of_nation WHERE nationkey = 3", 3, Optional.empty()},
+                {"UPDATE copy_of_nation SET name = 'POLAND' WHERE nationkey = 1", 2, Optional.of("This connector does not support updates")},
+                {"MERGE INTO copy_of_nation n USING region r ON r.regionkey= n.regionkey WHEN MATCHED THEN DELETE", 2, Optional.of("This connector does not support merges")},
+                {"DROP TABLE copy_of_nation", 3, Optional.empty()},
+                {"SHOW SCHEMAS", 1, Optional.empty()},
+                {"SHOW TABLES", 2, Optional.empty()},
+                {"SHOW STATS FOR nation", 2, Optional.empty()},
```

Comment:
nit: Is it possible to reuse the test cases to be shared by both h2 and PG connector ?